// @clerk/react/dist/internal.mjs
import {
  MultisessionAppSupport
} from "./chunk-Z2DSM5QN.mjs";
import {
  incompatibleRoutingWithPathProvidedError,
  noPathProvidedError,
  useDerivedAuth
} from "./chunk-PS5BRRRD.mjs";
import {
  errorThrower,
  setErrorThrowerOptions
} from "./chunk-MB46WFKC.mjs";
import "./chunk-OANWQR3B.mjs";

// src/hooks/useRoutingProps.ts
function useRoutingProps(componentName, props, routingOptions) {
  const path = props.path || (routingOptions == null ? void 0 : routingOptions.path);
  const routing = props.routing || (routingOptions == null ? void 0 : routingOptions.routing) || "path";
  if (routing === "path") {
    if (!path) {
      return errorThrower.throw(noPathProvidedError(componentName));
    }
    return {
      ...routingOptions,
      ...props,
      routing: "path"
    };
  }
  if (props.path) {
    return errorThrower.throw(incompatibleRoutingWithPathProvidedError(componentName));
  }
  return {
    ...routingOptions,
    ...props,
    path: void 0
  };
}

// src/internal.ts
import {
  clerkJsScriptUrl,
  buildClerkJsScriptAttributes,
  clerkUiScriptUrl,
  buildClerkUiScriptAttributes,
  setClerkJsLoadingErrorPackageName
} from "@clerk/shared/loadClerkJsScript";
export {
  MultisessionAppSupport,
  buildClerkJsScriptAttributes,
  buildClerkUiScriptAttributes,
  clerkJsScriptUrl,
  clerkUiScriptUrl,
  setClerkJsLoadingErrorPackageName,
  setErrorThrowerOptions,
  useDerivedAuth,
  useRoutingProps
};
